src/backend/marsha/core/tests/test_api_video_live_pairing.py

```diff
@@ -0,0 +1,354 @@
+        device = Device.objects.first()
+        self.assertEqual(device.id, box_id)
+
+    def test_api_video_pairing_challenge_valid_secret_jitsi_multiple_challenges(self):
+        """Multiple successfull pairing should not fail."""
+        video = factories.VideoFactory(live_state=IDLE, live_type=JITSI)
+        live_pairing = LivePairingFactory(video=video)
+
+        box_id = uuid.uuid4()
+        payload = {"box_id": box_id, "secret": live_pairing.secret}
+
+        # first sucessful pairing
+        response = self.client.post("/api/pairing-challenge", data=payload)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            {"jitsi_url": f"https://{settings.JITSI_DOMAIN}/{video.id}"},
+            response.json(),
+        )
+        self.assertEqual(LivePairing.objects.count(), 0)
+        self.assertEqual(Device.objects.count(), 1)
+        device = Device.objects.first()
+        self.assertEqual(device.id, box_id)
+
+        video = factories.VideoFactory(live_state=IDLE, live_type=JITSI)
+        live_pairing = LivePairingFactory(video=video)
+
+        payload = {"box_id": box_id, "secret": live_pairing.secret}
+
+        # second sucessful pairing on another video
+        response = self.client.post("/api/pairing-challenge", data=payload)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            {"jitsi_url": f"https://{settings.JITSI_DOMAIN}/{video.id}"},
+            response.json(),
+        )
+        self.assertEqual(LivePairing.objects.count(), 0)
+        self.assertEqual(Device.objects.count(), 1)
+        device = Device.objects.first()
+        self.assertEqual(device.id, box_id)
+
+    def test_api_video_pairing_challenge_unknown_devices_throttling(self):
+        """Throttling should prevent more than two requests per minute.
+
+        Possible scenarios:
+        - Multiple requests from legit box that were never been paired.
+        - Brute force attempt to guess a secret.
+        """
+        faker = Faker()
+        payload = {"box_id": uuid.uuid4(), "secret": "000000"}
+
+        # first 3 requests shouldn't be throttled
+        for _ in range(3):
+            response = self.client.post(
+                "/api/pairing-challenge",
+                data=payload,
+                REMOTE_ADDR=faker.ipv4(),
+            )
+            self.assertEqual(response.status_code, 404)
+            self.assertEqual(response.json(), {"detail": "Secret not found."})
+
+        # fourth request should be throttled
+        response = self.client.post(
+            "/api/pairing-challenge",
+            data=payload,
+            REMOTE_ADDR=faker.ipv4(),
+        )
+        self.assertEqual(response.status_code, 429)
+        self.assertEqual(
+            response.json(),
+            {"detail": "Request was throttled. Expected available in 60 seconds."},
+        )
+
+        # resetting throttling by mocking timer used by DRF in AnonRateThrottle
```

Comment:
calling `cache.clear()` is not easier ?

Reply:
Yes, it would be easier, but i think it's more realistic like this.

Reply:
Are you ok with this ?